### ISSUE TITLE
Timer time_left is wrong before it started

### ISSUE THREAD

**Issue by MaaaxiKing** [godotengine/godot#37942]
**Godot version:**
3.2.1
**OS/device including version:**
Windows 10 Home 1909
**Issue description:**
If you create a Timer in code, set its wait time and want to get it, it will return 0 before its start.
**Steps to reproduce:**
var timer = Timer.new()

func _ready():
	timer.set_wait_time(3)
	print(timer.time_left)
**Minimal reproduction project:**              I don't exactly understand what this means
var timer = Timer.new()

func _ready():
	timer.set_wait_time(3)
	add_child(timer)
	timer.one_shot = true
	print(timer.time_left)

**Comment by abreulima** [comment]
If a Timer isn't started, its wait_time is 0.

[https://docs.godotengine.org/en/3.2/classes/class_timer.html#class-timer-property-time-left](url)

**Comment by KoBeWi** [comment]
Yep, this is correct according to the docs.

**Comment by MaaaxiKing** [comment]
Sure it is, but in my view this is wrong.

**Comment by KoBeWi** [comment]
Changing this is probably trivial, but the current behavior makes sense. There's no time left if the timer isn't running, bruh.

If it's really important, I can reopen this for discussion (although technically it could go to [proposals repo](https://github.com/godotengine/godot-proposals)).

**Comment by MaaaxiKing** [comment]
I would like to hear opinions of others, I won't change my opinion. An example: if I set a timer on my phone for 1 minute, it will show me that the time left is 1 minute, even if the timer hasn't started yet.

**Comment by MWFIAE** [comment]
According to the godot principals it's not a bug as it's working as documented.
Also there is an easy workaround for this and probably it isn't an issue that many projects are facing. 

---

Workaround: 
``` 
var remaining_time := 0
if timer.is_stopped():
  remaining_time = timer.wait_time
else:
  remaining_time = timer.time_left
``` 
If it's something that needs to be done often in your project you can encapsulate the code above in a (static) function for easier use. 

---

However that's not saying that it shouldn't be changed. As I agree the behaviour is a little bit unusual. And there is probably not any code around that depends on this specific behaviour either.

**Comment by MaaaxiKing** [comment]
Imagine you have a sandglass with a little plate with hinges in the hole. If you want to start the sandglass, you have to press a lever which is at the edge of the sandglass. This makes the plate "fall down". Before you have started the sandglass/pressed the lever there was all sand in the top half, so this was the left time. I hope my example is comprehensible.

**Comment by svendixon** [comment]
I have to agree with @MaaaxiKing. If the wait time is the "amount of fuel I gave it" and the car hasn't started yet, I would expect the car to still have the fuel I gave it.

**Comment by amarraff** [comment]
I'm experiencing confusion on this as well in v4.3. I'm trying to set wait_time on ready, but the documentation doesn't make it clear whether this is possible. Is the only way to set wait_time prior to starting the timer via the node's inspector properties? I tried using my_timer.start(my_wait_time) as well, and that didn't work either. I'm not sure what I'm doing wrong. The timer will only use the pre-set wait_time in the inspector.

**Comment by KoBeWi** [comment]
`time_left` and `wait_time` are 2 different things.
Try making a minimal reproduction project for your problem so I can check it. Both `start(wait_time)` and setting `wait_time` before starting should overwrite whatever you set in the inspector. So either 1) your code that changes time does not run 2) you are changing different Timer.
Setting `wait_time` while the Timer is running has no effect. The doc could clarify that.

**Comment by amarraff** [comment]
> `time_left` and `wait_time` are 2 different things. Try making a minimal reproduction project for your problem so I can check it. Both `start(wait_time)` and setting `wait_time` before starting should overwrite whatever you set in the inspector. So either 1) your code that changes time does not run 2) you are changing different Timer. Setting `wait_time` while the Timer is running has no effect. The doc could clarify that.

I fixed it. The issue was that the variable I was using to set the wait_time was declared as an INT instead of a FLOAT, duh. So it was rounding my 0.3s wait time to 1s, the timer's default, making it look like wait time wasn't being set. Sorry, my bad. I swear I tried to figure this out for more than 2 seconds before I came here, lol.
